pyccl/tracers.py

```diff
@@ -0,0 +1,409 @@
+            `None`, the tracer is assumed to not have magnification bias
+            terms. Defaults to None.
+    """
+    def __init__(self, cosmo, has_rsd, dndz, bias, mag_bias=None):
+        self._trc = []
+
+        kernel_d = None
+        if bias is not None:  # Has density term
+            # Kernel
+            if kernel_d is None:
+                kernel_d = get_density_kernel(cosmo, dndz)
+            # Transfer
+            z_b, b = _check_array_params(bias)
+            # Reverse order for increasing a
+            t_a = (1./(1+z_b[::-1]), b[::-1])
+            self.add_tracer(cosmo, kernel=kernel_d, transfer_a=t_a)
+        if has_rsd:  # Has RSDs
+            # Kernel
+            if kernel_d is None:
+                kernel_d = get_density_kernel(cosmo, dndz)
+            # Transfer (growth rate)
+            z_b, _ = _check_array_params(dndz)
+            a_s = 1./(1+z_b[::-1])
+            t_a = (a_s, -growth_rate(cosmo, a_s))
+            self.add_tracer(cosmo, kernel=kernel_d,
+                            transfer_a=t_a, der_bessel=2)
+        if mag_bias is not None:  # Has magnification bias
+            # Kernel
+            chi, w = get_lensing_kernel(cosmo, dndz, mag_bias=mag_bias)
+            # Multiply by -2 for magnification
+            kernel_m = (chi, -2 * w)
+            self.add_tracer(cosmo, kernel=kernel_m,
+                            der_bessel=-1, der_angles=1)
+
+
+class WeakLensingTracer(Tracer):
+    """Specific `Tracer` associated to galaxy shape distortions including
+    lensing shear and intrinsic alignments within the L-NLA model.
+
+    Args:
+        cosmo (:obj:`Cosmology`): Cosmology object.
+        dndz (tuple of arrays): A tuple of arrays (z, N(z))
+            giving the redshift distribution of the objects. The units are
+            arbitrary; N(z) will be normalized to unity.
+        has_shear (bool): set to `False` if you want to omit the lensing shear
+            contribution from this tracer.
+        ia_bias (tuple of arrays, optional): A tuple of arrays
+            (z, A_IA(z)) giving the intrinsic alignment amplitude A_IA(z).
+            If `None`, the tracer is assumped to not have intrinsic
+            alignments. Defaults to None.
+    """
+    def __init__(self, cosmo, dndz, has_shear=True, ia_bias=None):
+        self._trc = []
+        if has_shear:
+            # Kernel
+            kernel_l = get_lensing_kernel(cosmo, dndz)
+            self.add_tracer(cosmo, kernel=kernel_l,
+                            der_bessel=-1, der_angles=2)
+        if ia_bias is not None:  # Has intrinsic alignments
+            z_a, tmp_a = _check_array_params(ia_bias)
+            # Kernel
+            kernel_i = get_density_kernel(cosmo, dndz)
+            # Normalize so that A_IA=1
+            D = growth_factor(cosmo, 1./(1+z_a))
+            # Transfer
+            rho_m = lib.cvar.constants.RHO_CRITICAL * cosmo['Omega_m']
+            a = - tmp_a * 5e-14 * rho_m / D
```

Comment:
We want a code comment explaining this magic number/formula with a reference to a paper.

Reply:
I'm actually not sure where this is coming from, @elisachisari 

If I go by Eq. 5.2 of https://arxiv.org/pdf/1506.02671.pdf, I believe this should be:
```
a = - tmp_a * cosmo['Omega_m'] * 0.0134 / D
```

Where does the 5e-14 come from?

Reply:
This is the original value of C_1 from SuperCOSMOS. C1*rho_crit=0.0134 in many works, but we have our own value of rho_crit we can use.

Reply:
Aaah, I see. OK, should we just add a comment with a citation to https://arxiv.org/abs/astro-ph/0009499 then?

Reply:
Yes, adding docs for this now